contentcuration/contentcuration/viewsets/user.py

```diff
@@ -266,7 +272,30 @@ def create_from_changes(self, changes):
 
     def delete_from_changes(self, changes):
         return self._handle_relationship_changes(changes)
-
+    @action(detail=True, methods=['delete'])
+    def remove_self(self, request, pk=None):
+        """
+        Allows a user to remove themselves from a channel as a viewer.
+        """
+        user = self.get_object()
+        channel_id = request.query_params.get('channel_id', None)
+
+        if not channel_id:
+            return HttpResponseBadRequest(_, 'Channel ID is required.')
+            
+
+        channel = Channel.objects.get(id=channel_id)
+        if not channel:
+            return HttpResponseNotFound("Channel not found {}".format(channel_id))
+
+        if request.user != user and not request.user.can_edit(channel_id):
+            return HttpResponseForbidden("You do not have permission to remove this user {}".format(user.id))
+
+        if channel.viewers.filter(id=user.id).exists():
+            channel.viewers.remove(user)
+            return HttpResponse(status=204)
```

Comment:
Let't not hard code the status. import the no content status code from

`from rest_framework.status import HTTP_204_NO_CONTENT` and then use `HTTP_204_NO_CONTENT` instead